WORKSPACE

```diff
@@ -0,0 +1,337 @@
+workspace(name = "com_google_closure_templates")
+
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+load("@bazel_tools//tools/build_defs/repo:java.bzl", "java_import_external")
+
+# TODO(yannic): Switch to Bazel-native dependencies when possible.
+
+# public domain
+java_import_external(
```

Comment:
would this workspace be easier to maintain if it used the rules_jvm_external? Then Bazel wouldn't be acting like a package manager - let maven do that.

https://blog.bazel.build/2019/03/31/rules-jvm-external-maven.html

Reply:
I'll look into it. Since every target is private for now, we can change how we declare dependencies later without breaking anyone.

Reply:
Yes, consider using rules_jvm_external. java_import_external is more tedious to maintain manually.

Reply:
I'll refrain from using `rules_jvm_external` for now because doing so would break `rules_closure` :(.

Reply:
I was just working on moving js-dossier to rules_jvm_external - it also uses rules_closure. What breakage did you run into?

Reply:
I didn't run directly into any breakage. The issue is that `rules_closure` declares it's dependencies using `java_import_external`. When we start using this Bazel-fied version from `rules_closure` (which is the reason I've been sending this PR), we will start depending on two different versions of, e.g. Guava (one from `rules_jvm_external` and one from `java_import_external`), which is obviousely bad (but unfortunately not failing the build by default in Bazel).

Reply:
Sorry just want to clarify: you want to avoid downloading jar files twice?

I don't see why it's necessarily a problem for rules_closure to use one mechanism to download one version of guava used in its own build actions, and for closure-templates to use a newer mechanism to download a different version of guava to build itself. It would make this PR cleaner, and we should be able to change rules_closure's dependency mechanism independently? OR maybe I misunderstand how this works still.

Reply:
Yes, I want to avoid downloading jars twice. But I'm also planning to integrate `closure-templates` into [ClosureWorker](https://github.com/bazelbuild/rules_closure/blob/master/java/io/bazel/rules/closure/ClosureWorker.java) to avoid the JVM startup penalty for every `.soy` file. If we do this, we'd link two versions of the same jar into one binary.

Also, I might be missing something, but how would I use `rules_jvm_external` in [rules_closure_dependencies](https://github.com/bazelbuild/rules_closure/blob/master/closure/repositories.bzl#L44)? AFAIK, we can't have load statements in macros, but we'd need them to call `maven_install` without forcing out users to import `rules_jvm_external` before calling `rules_closure_dependencies()`.

Reply:
IMO there should be a `java_library` with minimal deps that provides worker support (like https://npmjs.com/package/@bazel/worker for JS), and closure compiler should have a separate binary that acts as a worker from closure templates. Doesn't seem wise to me to make a binary that links them together.

I'll propose a separate plan for rules_closure over in that repo (changing current `rules_closure_dependencies` is a breaking change since users might rely on it to fetch jars that they reference) - my point here is just that my mental model is that closure-templates should declare and fetch its own dependencies and not rely on getting some of them through whatever mechanism rules_closure used